src/components/form/model-provider/model-provider.vue

```diff
@@ -0,0 +1,72 @@
+<script>
+    // check whether an input is a of kind toggle
+    const isToggle = target => target.getAttribute('type') === 'checkbox';
+
+    export default {
+        props: {
+            value: {
+                type: [String, Number, Array, Boolean],
+                required: true,
+            },
+        },
+        computed: {
+            model: {
+                get() {
+                    return this.value;
+                },
+                set(model) {
+                    this.$emit('input', model);
+                },
+            },
+        },
+        methods: {
+            updateValue({ target }) {
+                if (isToggle(target)) {
+                    this.updateToggle(target);
+                } else {
+                    this.model = target.value;
+                }
+            },
+            updateValueRaw(value) {
+                this.model = value;
+            },
+            updateToggle({ checked, value }) {
+                if (typeof this.value === 'boolean') {
+                    this.model = checked;
+                } else if (Array.isArray(this.model)) {
+                    if (!value) {
+                        throw new Error('I was unable to update the model. Please make sure your input has the value attribute set');
```

Comment:
Shouldn't the error message say something like this?
`Model could not be updated because there was no value attribute set on the input.`

Reply:
@nirazul much better now https://github.com/dreipol/vue-ui/pull/9/commits/b427dd7696c9369886738e769447dbe4c70bc49f